lib/puppet/provider/mysql_datadir/mysql.rb

```diff
@@ -6,7 +6,7 @@
   initvars
 
   # Make sure we find mysqld on CentOS
-  ENV['PATH']=ENV['PATH'] + ':/usr/libexec'
+  ENV['PATH']=ENV['PATH'] + ':/usr/libexec:/usr/sbin'
```

Comment:
Because it was already in your path. In some minimalist environment it can be absent.